substrate/primitives/arithmetic/src/per_things.rs

```diff
@@ -496,16 +496,35 @@ where
 	(x / maximum) * part_n + c
 }
 
+/// Checked compute of the error due to integer division in the expression `x / denom * numer`.
+fn rational_mul_correction<N, P>(x: N, numer: P::Inner, denom: P::Inner, rounding: Rounding) -> N
+where
+	N: MultiplyArg + UniqueSaturatedInto<P::Inner>,
+	P: PerThing,
+	P::Inner: Into<N>,
+{
+	checked_rational_mul_correction::<N, P>(x, numer, denom, rounding).unwrap()
+}
+
 /// Compute the error due to integer division in the expression `x / denom * numer`.
 ///
 /// Take the remainder of `x / denom` and multiply by  `numer / denom`. The result can be added
 /// to `x / denom * numer` for an accurate result.
-fn rational_mul_correction<N, P>(x: N, numer: P::Inner, denom: P::Inner, rounding: Rounding) -> N
+fn checked_rational_mul_correction<N, P>(
+	x: N,
+	numer: P::Inner,
+	denom: P::Inner,
+	rounding: Rounding,
+) -> Result<N, &'static str>
 where
 	N: MultiplyArg + UniqueSaturatedInto<P::Inner>,
 	P: PerThing,
 	P::Inner: Into<N>,
 {
+	if P::Inner::is_zero(&denom) {
+		return Err("Division by zero")
+	}
+
```

Comment:
In line `529` maybe there is a `checked_rem`? Then we dont need this if.

Reply:
not sure what you mean by `checked_rem`